compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -2409,3 +2409,83 @@ impl Visitor for RemoveGenericsAppearingInTypeVisitor<'_> {
         }
     }
 }
+
+/// Interpret source code using the elaborator, without
+/// parsing and compiling it with nargo, converting
+/// the result into a monomorphized AST expression.
+pub fn interpret(src: &str) -> crate::monomorphization::ast::Expression {
+    use crate::hir::{
+        Context, ParsedFiles,
+        def_collector::{dc_crate::DefCollector, dc_mod::collect_defs},
+        def_map::{CrateDefMap, ModuleData},
+    };
+    use crate::monomorphization::{Monomorphizer, debug_types::DebugTypeTracker};
+    use crate::parse_program;
+    use fm::{FileId, FileManager};
+    use std::path::PathBuf;
+
+    let file = FileId::default();
+
+    let location = Location::new(Default::default(), file);
+    let root_module = ModuleData::new(
+        None,
+        location,
+        Vec::new(),
+        Vec::new(),
+        false, // is contract
+        false, // is struct
+    );
+
+    let file_manager = FileManager::new(&PathBuf::new());
+    let parsed_files = ParsedFiles::new();
+    let mut context = Context::new(file_manager, parsed_files);
+    context.def_interner.populate_dummy_operator_traits();
+
+    let krate = context.crate_graph.add_crate_root(FileId::dummy());
+
+    let (module, errors) = parse_program(src, file);
+    // Skip warnings
+    let errors: Vec<_> = errors.iter().filter(|e| !e.is_warning()).collect();
+    assert_eq!(errors.len(), 0);
+    let ast = module.into_sorted();
+
+    let def_map = CrateDefMap::new(krate, root_module);
+    let root_module_id = def_map.root();
+    let mut collector = DefCollector::new(def_map);
+
+    collect_defs(&mut collector, ast, FileId::dummy(), root_module_id, krate, &mut context);
+    context.def_maps.insert(krate, collector.def_map);
+
+    let main = context.get_main_function(&krate).expect("Expected 'main' function");
+
+    let mut elaborator = Elaborator::elaborate_and_return_self(
+        &mut context,
+        krate,
+        collector.items,
+        ElaboratorOptions::test_default(),
+    );
+
+    // Skip compiler warnings
+    let errors: Vec<_> = elaborator.errors.iter().filter(|&e| e.is_error()).cloned().collect();
+    if !errors.is_empty() {
+        log::debug!("elaborator errors: {:?}", errors);
+    }
+    assert_eq!(errors.len(), 0);
```

Comment:
Return errors instead of logging and panicking? Now that it's a public function, it would be nice to behave like the others.